presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -0,0 +1,342 @@
+                validateShowStatsWhereExpression(((LogicalBinaryExpression) expression).getRight(), node);
+            }
+            else if (expression instanceof ComparisonExpression) {
+                validateShowStatsWhereExpression(((ComparisonExpression) expression).getLeft(), node);
+                validateShowStatsWhereExpression(((ComparisonExpression) expression).getRight(), node);
+            }
+            else if (expression instanceof IsNullPredicate) {
+                validateShowStatsWhereExpression(((IsNullPredicate) expression).getValue(), node);
+            }
+            else if (expression instanceof IsNotNullPredicate) {
+                validateShowStatsWhereExpression(((IsNotNullPredicate) expression).getValue(), node);
+            }
+        }
+
+        private Node rewriteShowStats(ShowStats node, Table table, Constraint<ColumnHandle> constraint)
+        {
+            TableHandle tableHandle = getTableHandle(node, table.getName());
+            TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
+            List<String> statisticsNames = findUniqueStatisticsNames(tableStatistics);
+            List<String> resultColumnNames = buildColumnsNames(statisticsNames);
+            List<SelectItem> selectItems = buildSelectItems(resultColumnNames);
+            Map<ColumnHandle, String> columnNames = getStatisticsColumnNames(tableStatistics, node, table.getName());
+
+            List<Expression> resultRows = buildStatisticsRows(tableStatistics, columnNames, statisticsNames);
+
+            return simpleQuery(selectAll(selectItems),
+                    aliased(new Values(resultRows),
+                            "table_stats_for_" + table.getName(),
+                            resultColumnNames));
+        }
+
+        private static void check(boolean condition, ShowStats node, String message)
+        {
+            if (!condition) {
+                throw new SemanticException(NOT_SUPPORTED, node, message);
+            }
+        }
+
+        @Override
+        protected Node visitNode(Node node, Void context)
+        {
+            return node;
+        }
+
+        private Constraint<ColumnHandle> getConstraint(QuerySpecification specification)
+        {
+            if (!specification.getWhere().isPresent()) {
+                return Constraint.alwaysTrue();
+            }
+
+            Plan plan = queryExplainer.get().getLogicalPlan(session, new Query(Optional.empty(), specification, Optional.empty(), Optional.empty()), parameters);
+
+            Optional<TableScanNode> scanNode = searchFrom(plan.getRoot())
+                    .where(TableScanNode.class::isInstance)
+                    .findFirst();
+
+            if (!scanNode.isPresent()) {
+                return new Constraint<>(TupleDomain.none(), bindings -> true);
+            }
+
+            Expression filterNodePredicate = searchFrom(plan.getRoot())
```

Comment:
I added a fixup commit which removes special handling for `FilterNode` in `ShowStatsRewrite`.